packages/flutter_tools/test/general.shard/plugins_test.dart

```diff
@@ -612,6 +612,61 @@ web_plugin_with_nested:${webPluginWithNestedFile.childDirectory('lib').uri.toStr
         FeatureFlags: () => featureFlags,
       });
     });
+
+    testUsingContext('dev_dependencies aren\'t included in plugins', () async {
```

Comment:
This is a great start, but I would want a few more test cases. Some examples:

- What if a direct and dev dependency have a common ancestor?
- Edge case: pubspec.
- Edge case: invalid dependencies
- Edge case: no direct dependencies

et cetera